pepper-apis/src/main/java/org/broadinstitute/ddp/filter/LanguageResolutionFilter.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.ddp.filter;
+
+import static spark.Spark.halt;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Locale.LanguageRange;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+
+import org.broadinstitute.ddp.constants.RouteConstants;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.StudyDao;
+import org.broadinstitute.ddp.util.I18nUtil;
+import org.broadinstitute.ddp.util.RouteUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import spark.Filter;
+
+import spark.Request;
+import spark.Response;
+
+/**
+ * Figures out a preferred user language taking into account the language weights
+ * from Accept-Language header, information from the user profile and languages
+ * supported by the study. Puts that language into the attribute storage to make
+ * it available later in all routes interested in fetching translated entities
+ */
+public class LanguageResolutionFilter implements Filter {
+
+    public static final String USER_LANGUAGE = "USER_LANGUAGE";
+    private static final Logger LOG = LoggerFactory.getLogger(LanguageResolutionFilter.class);
+
+    @Override
+    public void handle(Request request, Response response) {
+        try {
+            String acceptLanguageHeader = request.headers(RouteConstants.ACCEPT_LANGUAGE);
+            Matcher matcher = Pattern.compile("/studies/(\\w+)").matcher(request.url());
```

Comment:
`RouteUtil.STUDIES_PATH_MARKER` is the place to look for `studies`--can you change to refer to that?

Reply:
Actually, a nice refactor would be to clean up `RouteUtil.STUDIES_PATH_MARKER` and `RouteConstants.API.STUDY_ALL` and make `"studies"` more D-R-Y.